CMake/External_GDAL.cmake

```diff
@@ -74,9 +74,21 @@ else()
     # from macports.  GDAL's '--with-libiconv-prefix' option looks like it should handle
     # this but in fact seems to do nothing.
     #
-    set(_GDAL_ARGS_APPLE --without-libtool --with-netcdf=no --with-curl=no --with-local=/usr)
+    # Note: previously this var disabled curl and netcdf which are no handled
+    # by _GDAL_ARGS_UNSUPPORTED
+    set(_GDAL_ARGS_APPLE --without-libtool --with-local=/usr)
   endif()
 
+  # GDAL uses a configure based build system, so its important to disable
+  # anything that is not explicitly built or provided through fletch. If these
+  # are not disabled then it may find a system version of these libs. If the
+  # system also includes a conflicting version of another library provided by
+  # fletch, that can cause errors. For example, if you have a conda environment
+  # with curl and proj, just by adding curl to the include search paths, GDAL
+  # will ignore the fletch version of proj (even though we do specify it
+  # explicitly here) and use the conda version.
+  set(_GDAL_ARGS_UNSUPPORTED --with-curl=no --with-netcdf=no --with-kea=no --with-geos=no)
```

Comment:
GEOS is explicitly built in Fletch. I don't know offhand what role it plays in GDAL but It would be worth understanding before shutting it off. The others, I agree with.

That said, one could do if (Fletch_ENABLE_GEOS) and set  --with-geos=${fletch_BUILD_INSTALL_PREFIX}/bin/geos-config ... if it's there.

Reply:
I'll look into what GEOS does in fletch. It seems like it would be be ok to use fletch's version. 

I think with libtiff, its not that gdal ignore's the explicit path (assuming that its the same issues as the one I was investigating here), its that there is some other library that's included which happens to exist in a directory that a different version of libtiff also exists in. Because all of the import paths are declared globally, there is no way to differentiate between multiple different libtiff files if there happens to be another version in another included library. The way to test this is to determine if there are any non-fletch libraries being included that happen to exist parallel to another libtiff. From what I understand, if all dependencies libraries existed in fletch, then only the fletch install include and lib directories would be searched and the problem would be "fixed".

Reply:
For reference: GEOS (Geometry Engine Open Source) implements geometry data structures and operations and serializations, notably WTK and WKB formats. It handles important nuances of topological intersections / unions / etc... 

When GEOS is disabled in GDAL it looks like it might cause some [funky](https://github.com/OSGeo/gdal/blob/595fc490ff5c1572fd1e75b4d5840ef30a170e74/gdal/ogr/ogrgeometry.cpp#L488) behavior: https://www.pythonanywhere.com/forums/topic/2801/ and [disable](https://github.com/OSGeo/gdal/blob/595fc490ff5c1572fd1e75b4d5840ef30a170e74/gdal/ogr/ogrgeometry.cpp#L2121) certain (probably important) functionalities. 

I don't think any of the image-related gdal functionality would be impacted, but it would lose the ability to handle complex geometrical operations. I'm not sure how important that is for KWIVER's usage of gdal in general, but I know I use those features of gdal in my own work. 

So disabling geos seems to be a bad idea.